src/util/permissions.tsx

```diff
@@ -114,36 +118,28 @@ export const getResourceTypeOptions = (metadata?: LxdMetadata | null) => {
     options.push({
       value: resourceType,
       label: label.join(" "),
+      text: label.join(" "),
     });
   }
 
   return options;
 };
 
-const sortOptions = (
-  a: OptionHTMLAttributes<HTMLOptionElement>,
-  b: OptionHTMLAttributes<HTMLOptionElement>,
-): number => {
-  if (b.label === "Select an option") {
-    return 1;
-  }
-
-  return (a.label ?? "").localeCompare(b.label as string);
+const sortOptions = (a: CustomOption, b: CustomOption): number => {
+  return (a.text ?? a.value).localeCompare(b.text ?? b.value);
```

Comment:
I would keep this outside of the custom select component since sorting is not currently part of its responsibilities. Unless we want to include a sort option for the custom select?